service_identity/src/main/kotlin/org/thechance/service_identity/data/geteway/UserGateWayImp.kt

```diff
@@ -0,0 +1,52 @@
+package org.thechance.service_identity.data.geteway
+
+import com.mongodb.client.model.IndexOptions
+import com.mongodb.client.model.Indexes
+import org.bson.types.ObjectId
+import org.koin.core.annotation.Single
+import org.litote.kmongo.eq
+import org.litote.kmongo.setTo
+import org.litote.kmongo.set
+import org.thechance.service_identity.data.DataBaseContainer
+import org.thechance.service_identity.data.collection.UserCollection
+import org.thechance.service_identity.domain.entity.User
+import org.thechance.service_identity.domain.gateway.UserGateWay
+import org.thechance.service_identity.utils.Constants.USER_COLLECTION
+
+@Single
+class UserGateWayImp(dataBaseContainer: DataBaseContainer): UserGateWay {
+
+    private val userCollection by lazy { dataBaseContainer.database.getCollection<UserCollection>(
+        USER_COLLECTION
+    ) }
+
+    override suspend fun getUserById(id: String): User? =
+        userCollection.findOne(
+            UserCollection::id eq ObjectId(id),
+            UserCollection::isDeleted eq false
+        )?.toUser()
+
+    override suspend fun getUsers(): List<User> =
+        userCollection.find(
+            UserCollection::isDeleted eq false
+        ).toList().toUser()
+
+    override suspend fun createUser(user: User): Boolean {
+        val indexOptions = IndexOptions().unique(true)
+        userCollection.createIndex(Indexes.ascending("username"), indexOptions)
+        return userCollection.insertOne(user.toUserCollection()).wasAcknowledged()
+    }
```

Comment:
I noticed a potential issue here. It seems that we're creating an index every time we create a user, which could be costly. Could you check with your teammates if there's a way to create the index once, like on a schema? or by moving the index creation outside?

@NadaFeteiha

Reply:
I knew I was leaving it for the meeting today, so I left it there just in case someone has a solution for it 😂

Reply:
Well done Youseef👏